Assets/BossRoom/Scripts/Client/UI/IPConnectionWindow.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using System.Collections;
+using TMPro;
+using Unity.Netcode;
+using Unity.Netcode.Transports.UNET;
+using Unity.Netcode.Transports.UTP;
+using UnityEngine;
+
+namespace Unity.Multiplayer.Samples.BossRoom.Visual
+{
+    public class IPConnectionWindow : MonoBehaviour
+    {
+        [SerializeField]
+        CanvasGroup m_CanvasGroup;
+
+        [SerializeField]
+        IPUIMediator m_IPUIMediator;
```

Comment:
Shouldn't a reference to this object be passed in via DI?

Reply:
SerializeField is a form of DI by itself, just controlled by the editor

Reply:
The reference would always be fulfilled and could never fail, unless there is no IPUIMediator bound (which in itself is a programmer error).

Reply:
Since we are using DI for similar purposes elsewhere - I think that it would also be idiomatic and conformant to the general approach we are taking.

Reply:
After discussion: will pass in via DI.